app.rb

```diff
@@ -0,0 +1,121 @@
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    @people.each do |person|
+      if person.instance_of?(Student)
+        puts "[Student], ID: #{person.id}, Name: #{person.name}, age: #{person.age}"
+      else
+        puts "[Teacher], ID: #{person.id}, Name: #{person.name}, age: #{person.age}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to add a student (1) or a teacher (2)? [Input the number]: '
+    student_or_teacher = gets.chomp.to_i
+    case student_or_teacher
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Please enter a valid number: 1 or 2'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    puts 'Has parent permission? [Y/N]'
+    permission = gets.chomp.downcase == 'y'
+    student = Student.new(age, permission, name)
+    @people << student
+    puts 'Student created sucessfully'
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    puts 'specialization'
+    specialization = gets.chomp
+    teacher = Teacher.new(age, specialization, name)
+    @people << teacher
+    puts 'Teacher created sucessfully'
+  end
+
+  def add_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created sucessfully'
+  end
+
+  def add_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Auhtor: #{book.author}"
+    end
+    book_idx = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_idx = gets.chomp.to_i
+
+    print 'Date: '
+    rental_date = gets.chomp
+
+    rental = Rental.new(rental_date, @books[book_idx], @people[person_idx])
+    @rentals << rental
+    puts 'Rental created successfully'
+  end
+
+  # def list_all_rentals
+  #   print 'ID of person: '
+  #   id = gets.chomp
+  #   puts 'Rental: '
+  #   @rentals.each do |rental|
+  #     puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}" if rental.person.id == id
+  #   end
+  # end
+  # Method to list all rentals for a given person ID
+  def list_all_rentals
+    print 'ID of person: '
+    id = gets.chomp
+    puts 'Rentals: '
+    matching_rentals = @rentals.select { |rental| rental.person.id == id if rental.person.respond_to?(:id) }
+
+    if matching_rentals.empty?
+      puts 'No rentals found for this person ID.'
+    else
+      matching_rentals.each do |rental|
+        puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}"
+      end
+    end
```

Comment:
- [ ] **Good job on fixing the error in listing the rentals 👏 👏** However, it would be better to list the rentals as shown in the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ). Currently, I am creating the rentals but they are not being listed 😨  I have recorded a video to demonstrate 🙏 

https://www.loom.com/share/6304b0b61ad14993a712383d97eadd2d?sid=a9f7c364-f0c0-4500-ba30-5a6d5909c1f8